json_object.c

```diff
@@ -1265,3 +1265,128 @@ int json_object_equal(struct json_object* jso1, struct json_object* jso2)
 	return 0;
 }
 
+static int json_object_copy_serializer_data(struct json_object *src, struct json_object *dst)
+{
+	/* `src` and `dst` were validated by now ==> no null check for them */
+	if (src->o_type != json_type_double)
+		return 0;
+	/* Nothing to do */
+	if (!src->_userdata)
+		return 0;
+	if (!src->_user_delete) {
+		errno = EINVAL;
+		return -1;
+	}
+	/* we don't know how to handle other custom serializers (yet) */
+	if (json_object_userdata_to_json_string != src->_to_json_string)
+		return 0;
+	if (!(dst->_userdata = strdup((char *)src->_userdata))) {
+		errno = ENOMEM;
+		return -1;
+	}
+	dst->_to_json_string = json_object_userdata_to_json_string;
+	dst->_user_delete = src->_user_delete;
+	return 0;
+}
+
+static int json_object_deep_copy_recursive(struct json_object *src, struct json_object **dst)
+{
+	struct json_object *jso = NULL;
+	struct json_object_iter iter;
+	size_t i;
+	int rc = 0;
+
+	if (!src || !dst) {
+		errno = EINVAL;
+		return -1;
+	}
+
+	switch (src->o_type) {
+		case json_type_boolean:
+			*dst = json_object_new_boolean(src->o.c_boolean);
+			break;
+
+		case json_type_double:
+			*dst = json_object_new_double(src->o.c_double);
+			break;
+
+		case json_type_int:
+			*dst = json_object_new_int64(src->o.c_int64);
+			break;
+
+		case json_type_string:
+			*dst = json_object_new_string(get_string_component(src));
+			break;
+
+		case json_type_object:
+			*dst = json_object_new_object();
+			if (!*dst)
+				return -1;
+			json_object_object_foreachC(src, iter) {
+				/* This handles the `json_type_null` case */
+				if (!iter.val) {
+					json_object_object_add(*dst, iter.key, NULL);
+					continue;
+				}
+				if ((rc = json_object_deep_copy_recursive(iter.val, &jso)) < 0)
```

Comment:
Did you intend to do something with rc before returning?  If not, don't bother using it.  It looks like the only place it's actually used at all is right at the end of this function, and even there it's not really needed.